src/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

```diff
@@ -95,16 +122,24 @@ public async Task<InstanceDiscoveryMetadataEntry> GetMetadataEntryAsync(string a
                         return entry;
                     }
 
-                    entry = await FetchNetworkMetadataOrFallbackAsync(requestContext, authorityUri, environment).ConfigureAwait(false);
-
-                    if (entry != null)
+                    // If the user provides metadata, don't go to the network
+                    if (_userMetadataProvider != null)
                     {
-                        return entry;
+                        entry = _userMetadataProvider.GetMetadata(environment);
                     }
+                    else
```

Comment:
perhaps the priority selection should be extracted to a different class? that has that responsibility?

Reply:
The responsability of this class is to call providers in the correct order.